### PR TITLE
Make PointerPoint and PointerPointProperties to be a struct.

### PR DESCRIPTION
**Pull request by Takoooooo**
## Fixed issues
Part of #6666

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0023248-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: samples/ControlCatalog/Pages/PointerCanvas.cs [hunk 1]
```diff
@@ -132,20 +132,20 @@ protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
...
+                            IsMiddleButtonPressed: {_lastProperties?.IsMiddleButtonPressed}
+                            IsXButton1Pressed: {_lastProperties?.IsXButton1Pressed}
+                            IsXButton2Pressed: {_lastProperties?.IsXButton2Pressed}
+                            IsBarrelButtonPressed: {_lastProperties?.IsBarrelButtonPressed}
+                            IsEraser: {_lastProperties?.IsEraser}
+                            IsInverted: {_lastProperties?.IsInverted}
+                            Pressure: {_lastProperties?.Pressure}
+                            XTilt: {_lastProperties?.XTilt}
+                            YTilt: {_lastProperties?.YTilt}
+                            Twist: {_lastProperties?.Twist}";
```
Comment (hez2010) [inline samples/ControlCatalog/Pages/PointerCanvas.cs:148]: I think this formatting is incorrect. It will actually insert blankspaces starting with line 2.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0023252-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by danwalmsley — APPROVED** [review]

**Review by kekekeks — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0023261-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**